File: src/main/java/codes/qdbp/serverplugin/listeners/InventoryListener.java
package codes.qdbp.serverplugin.listeners;

import codes.qdbp.serverplugin.Serverplugin;
import codes.qdbp.serverplugin.inventories.PlayerDeathInventory;
import codes.qdbp.serverplugin.misc.Upgrades;
import net.kyori.adventure.text.Component;
import org.bukkit.Material;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.inventory.InventoryClickEvent;
import org.bukkit.inventory.ItemStack;

import java.util.UUID;


public class InventoryListener implements Listener {

    private final Serverplugin plugin;


    public InventoryListener(Serverplugin plugin) {
        this.plugin = plugin;
    }

    @EventHandler
    public void onInventoryClick(InventoryClickEvent event) {
        if (event.getCurrentItem() == null) return;

        Player player = (Player) event.getWhoClicked();
        Component inventoryTitle = event.getView().title().compact();

        if (inventoryTitle.equals(Component.text("--- Death Overview"))) {
            event.setCancelled(true);
            handleDeathInventory(player, event.getCurrentItem());
        } else if (inventoryTitle.toString().contains(" deaths")) {
            event.setCancelled(true);
        }
    }


    private void handleDeathInventory(Player player, ItemStack clickedItem) {
        player.closeInventory();
        player.openInventory(new PlayerDeathInventory(clickedItem, plugin).getPlayerDeathInventory());
    }




}

File: src/main/kotlin/codes/qdbp/serverplugin/listeners/InventoryListener.kt
package codes.qdbp.serverplugin.listeners

import codes.qdbp.serverplugin.Serverplugin
import codes.qdbp.serverplugin.menusystem.menus.PlayersDeathsMenu
import codes.qdbp.serverplugin.misc.menuManager
import codes.qdbp.serverplugin.misc.string
import net.kyori.adventure.text.serializer.plain.PlainTextComponentSerializer
import org.bukkit.Material
import org.bukkit.enchantments.Enchantment
import org.bukkit.entity.Player
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.inventory.InventoryClickEvent

class InventoryListener(val plugin: Serverplugin) : Listener {

    @EventHandler
    fun onInventoryClick(event: InventoryClickEvent) {
        if (event.currentItem == null) return
        val player = event.whoClicked as Player

        val inventoryTitle = PlainTextComponentSerializer.plainText().serialize(event.view.title())

        when {
            inventoryTitle.contains("Deaths") -> {
                event.isCancelled = true
            }

            inventoryTitle == "Change Efficiency" -> {
                event.isCancelled = true
                player.inventory.itemInMainHand.editMeta { meta ->
                    meta.addEnchant(Enchantment.DIG_SPEED, PlainTextComponentSerializer.plainText().serialize(event.currentItem!!.displayName()).removeSurrounding("[", "]").toIntOrNull() ?: 0, true)
                }
            }
        }
    }
}